Core/GDCore/IDE/Events/EventsVariablesFinder.cpp

```diff
@@ -203,6 +207,32 @@ std::set<gd::String> EventsVariablesFinder::FindArgumentsInInstructions(
   return results;
 }
 
+std::set<gd::String> EventsVariablesFinder::FindArgumentsInEventsAndDependencies(
+    const gd::Platform& platform,
+    const gd::Project& project,
+    const gd::Layout& layout,
+    const gd::EventsList& events,
+    const gd::String& parameterType,
+    const gd::String& objectName) {
+  std::set<gd::String> results;
+
+  std::set<gd::String> results2 = FindArgumentsInEvents(
+      platform, project, layout, events, parameterType, objectName);
+  results.insert(results2.begin(), results2.end());
+
+  DependenciesAnalyzer dependenciesAnalyzer = DependenciesAnalyzer(project, layout);
+  dependenciesAnalyzer.Analyze();
+  for (gd::String externalEventName : dependenciesAnalyzer.GetExternalEventsDependencies()) {
+    const gd::ExternalEvents& externalEvents = project.GetExternalEvents(externalEventName);
+
+    std::set<gd::String> results3 = FindArgumentsInEvents(
+        platform, project, layout, externalEvents.GetEvents(), parameterType, objectName);
+    results.insert(results3.begin(), results3.end());
+  }

```

Comment:
While now heavily discouraged, it's still possible to have a link in events linking to a scene. So we should also iterate on the scenes returned by `GetScenesDependencies` (doing the same work as for external events)